### PR TITLE
New tests to do file and directory sharing 

### PR DESCRIPTION
**Pull request by jnfrmarks**
I've created two new tests that test file and directory sharing.  

These tests validate:

sharing for users and users in groups that include:
resharing files and directories
editing files
deleting files
adding files
unsharing files and directories

### CODE REVIEW

File: python/smashbox/utilities/__init__.py [hunk 1]
```diff
@@ -120,13 +157,12 @@ def run_ocsync(local_folder,remote_folder="",N=None):
 
     for i in range(N):
         t0 = datetime.datetime.now()
-        cmd = config.oc_sync_cmd+' '+local_folder+' '+oc_webdav_url('owncloud',remote_folder)+" >> "+ config.rundir+"/%s-ocsync.step%02d.cnt%03d.log 2>&1"%(reflection.getProcessName(),current_step,ocsync_cnt[current_step])
+        cmd = config.oc_sync_cmd+' '+local_folder+' '+oc_webdav_url(config.oc_root,remote_folder,userNum)+" >> "+ config.rundir+"/%s-ocsync.step%02d.cnt%03d.log 2>&1"%(reflection.getProcessName(),current_step,ocsync_cnt[current_step])
```
Comment (moscicki) [inline python/smashbox/utilities/__init__.py:160]: config.oc_root should not be used here. The first argument of oc_webdav_url is the specification of internal protocol name (http or owncloud) used by owncloudcmd or other client tools

File: python/smashbox/utilities/__init__.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 ######## TEST SETUP AND PREPARATION
 
-def reset_owncloud_account(reset_procedure=None):
+def reset_owncloud_account(reset_procedure=None,numTestUsers=None):
```
Comment (moscicki) [inline python/smashbox/utilities/__init__.py:12]: It is impossible to specify that there is only one account which  name should not be altered. I propose to change it in the follow way: value None should mean that there is only one account name as specified by oc_account_name. Any other value (number) means that we enter the numbered account mode and that the number will be appended to oc_account_name. In the test code if you want to use the default number of users you should specify that explicitly: (...., numTestUsers=config.oc_number_test_users)

How does it sound for you?

File: python/smashbox/utilities/__init__.py [hunk 3]
```diff
@@ -97,13 +131,16 @@ def oc_webdav_url(protocol='http',remote_folder=""):
...
+
+  username = "%s%i"%(config.oc_account_name, userNum)
+  return protocol + '://' + username + (':%(oc_account_password)s@%(oc_server)s/'%config)+remote_path
 
 
 # this is a local variable for each worker that keeps track of the repeat count for the current step
 ocsync_cnt = {}
 
-def run_ocsync(local_folder,remote_folder="",N=None):
+def run_ocsync(local_folder,remote_folder="",N=None,userNum=1):
```
Comment (moscicki) [inline python/smashbox/utilities/__init__.py:143]: Same comment about userNum as for reset_owncloud_account() function.

File: python/smashbox/utilities/__init__.py [hunk 4]
```diff
@@ -158,12 +194,15 @@ def runcmd(cmd,ignore_exitcode=False,echo=True,allow_stderr=True,shell=True):
                 logger.error("stderr: %s",stderr)
 
     if process.returncode != 0:
-        msg = "Non-zero exit code %d from command %s" % (ignore_exitcode,repr(cmd))
+        msg = "Non-zero exit code %d from command %s" % (process.returncode,repr(cmd))
```
Comment (moscicki) [inline python/smashbox/utilities/__init__.py:197]: is this a bugfix?

File: etc/smashbox.conf.template [hunk 5]
```diff
@@ -23,6 +26,7 @@ oc_account_password=""
 # owncloud test server
 # if left blank or "localhost" then the real hostname of the localhost will be set
 oc_server = ''
+oc_root = 'owncloud'
```
Comment (moscicki) [inline etc/smashbox.conf.template:29]: We should see how oc_root interplays with oc_webdav_endpoint. Probably oc_webdav_endpoint should be defined as "%(oc_root)s/remote.php/webdav". Also please note that oc_root (which is a part of visible URL) for many installation is just and empty string. Please check if this usage is compatible with your concept of this configuration variable.

Comment (jnfrmarks) [inline etc/smashbox.conf.template:29]: Thanks!  I wasn't sure how to add oc_root to properties being defined in the conf file.  I'll take a look at making this change.

File: python/smashbox/utilities/__init__.py (hunk 2 above)
Comment (jnfrmarks) [inline python/smashbox/utilities/__init__.py:12]: I like this suggestion.  I'll make the code changes and test.

File: python/smashbox/utilities/__init__.py (hunk 3 above)
Comment (jnfrmarks) [inline python/smashbox/utilities/__init__.py:143]: Got it :)

File: python/smashbox/utilities/__init__.py (hunk 1 above)
Comment (jnfrmarks) [inline python/smashbox/utilities/__init__.py:160]: I need to look into why I made this change and will get back to you on why I made this change.

File: python/smashbox/utilities/__init__.py (hunk 4 above)
Comment (jnfrmarks) [inline python/smashbox/utilities/__init__.py:197]: Same as above: I need to look into why I made this change.

**Comment by jnfrmarks** [comment]
I'm moving this code to a different repo so I'm closing this PR.  It is superseded by PR #2 

I incorporated all suggested changes there.

**Comment by moscicki** [comment]
Thanks! I will be away for one week. You will here from me after than.

kuba

##

**Pull request closed without merging**